### PR TITLE
feat: close card detail with a downward swipe

### PR DESCRIPTION
**Pull request by Omochice**
## Summary

Lets the card detail bottom sheet be dismissed with a downward swipe on touch devices.

## Details

The sheet could previously only be closed via the backdrop, close button, or Escape, none of which match the expected dismissal gesture for a mobile bottom sheet.

A touch-only Svelte action handles the drag, engaging only when the content is scrolled to the top so it never hijacks scrolling, and leaving the sheet translated on release so the existing exit transition continues the motion.

## Confirmation

Ran the unit tests (6 passing), svelte-check (0 errors, 0 warnings), and biome (clean) locally.

## Limitation

The gesture is intentionally touch-only; pointer and keyboard users continue to rely on the backdrop, close button, or Escape.

## Summary by CodeRabbit

* **New Features**
  * Added swipe-down-to-dismiss gesture for modal/detail cards on touch devices.

* **Improvements**
  * Modal implementation updated to use the browser dialog with improved backdrop behavior, consistent Escape/backdrop close handling, and page-scroll lock while open.

* **Tests**
  * Added tests covering swipe-to-dismiss gestures and edge cases.

### CODE REVIEW

File: packages/site/src/lib/utils/swipe-down-to-dismiss.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+          node.style.transition = "";
+        }
+      }, RESET_TRANSITION_MS);
+    } else {
+      node.style.transition = "";
+    }
+    node.style.transform = "";
+    state.dragging = false;
+    state.deltaY = 0;
+  }
```
[the remark this answers is not in the document]
Comment (Omochice) [reply packages/site/src/lib/utils/swipe-down-to-dismiss.ts:59]: fixed in 5110d80

File: packages/site/src/lib/utils/swipe-down-to-dismiss.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  function onTouchStart(event: TouchEvent): void {
+    if (node.scrollTop > 0 || event.touches.length !== 1) {
+      return;
+    }
+    state.startX = event.touches[0].clientX;
+    state.startY = event.touches[0].clientY;
+    state.deltaY = 0;
+    state.dragging = true;
+    node.style.transition = "none";
+  }
```
[the remark this answers is not in the document]
Comment (Omochice) [reply packages/site/src/lib/utils/swipe-down-to-dismiss.ts:78]: fixed in 5110d80

File: packages/site/src/lib/utils/swipe-down-to-dismiss.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  return {
+    update(next: SwipeDownToDismissOptions): void {
+      current = next;
+    },
+    destroy(): void {
+      node.removeEventListener("touchstart", onTouchStart);
+      node.removeEventListener("touchmove", onTouchMove);
+      node.removeEventListener("touchend", onTouchEnd);
+      node.removeEventListener("touchcancel", onTouchCancel);
+    },
```
[the remark this answers is not in the document]
Comment (Omochice) [reply packages/site/src/lib/utils/swipe-down-to-dismiss.ts:139]: fixed in 5110d80

File: packages/site/src/lib/utils/swipe-down-to-dismiss.ts [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+      node.style.transition = "";
+    }
+    node.style.transform = "";
+    state.dragging = false;
+    state.deltaY = 0;
+  }
+
+  function onTouchCancel(): void {
+    resetDrag(true);
+  }
```
[the remark this answers is not in the document]
Comment (Omochice) [reply packages/site/src/lib/utils/swipe-down-to-dismiss.ts:66]: fixed in 5eb0079

File: packages/site/src/lib/CardDetail.svelte [hunk 5]
```diff
@@ -24,144 +25,167 @@
...
+		border: none;
 		border-radius: 24px 24px 0 0;
 		width: 100%;
 		max-width: 48rem;
 		max-height: 80vh;
 		overflow-y: auto;
+		/* Stop a swipe or scroll at the sheet edge from chaining to the page. */
+		overscroll-behavior: contain;
 		padding-top: 12px;
 	}
```
[the remark this answers is not in the document]
Comment (Omochice) [reply packages/site/src/lib/CardDetail.svelte:189]: fixed in 7370a5d

File: packages/site/src/lib/utils/swipe-down-to-dismiss.ts [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+    node.style.transition = "none";
+  }
+
+  function onTouchMove(event: TouchEvent): void {
+    if (!state.dragging) {
+      return;
+    }
+
+    const deltaX = event.touches[0].clientX - state.startX;
+    const deltaY = event.touches[0].clientY - state.startY;
```
[the remark this answers is not in the document]
Comment (Omochice) [reply packages/site/src/lib/utils/swipe-down-to-dismiss.ts:93]: fixed in a08f61d

File: packages/site/src/lib/CardDetail.svelte [hunk 7]
```diff
@@ -24,142 +25,166 @@
...
-
-	function handleKeydown(e: KeyboardEvent) {
-		if (e.key === "Escape") {
+	// A click whose target is the dialog itself landed on the (transparent)
+	// native backdrop rather than the sheet content, so treat it as dismiss.
+	function handleDialogClick(event: MouseEvent) {
+		if (event.target === dialogRef) {
 			onClose();
 		}
 	}
```
[the remark this answers is not in the document]
Comment (Omochice) [reply packages/site/src/lib/CardDetail.svelte]: fixed in 16a882f

File: packages/site/src/lib/utils/swipe-down-to-dismiss.test.ts [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+        ],
+        cancelable: true,
+        bubbles: true,
+      }),
+    );
+    node.dispatchEvent(touchEvent("touchend", { clientX: 0, clientY: 250 }));
+
+    expect(onDismiss).not.toHaveBeenCalled();
+    expect(node.style.transform).toBe("");
+  });
```
[the remark this answers is not in the document]
Comment (Omochice) [reply packages/site/src/lib/utils/swipe-down-to-dismiss.test.ts:85]: fixed in cfe9744

File: packages/site/src/lib/CardDetail.svelte [hunk 9]
```diff
@@ -24,142 +25,173 @@
...
+		const rect = dialogRef.getBoundingClientRect();
+		const isOutsideSheet =
+			event.clientX < rect.left ||
+			event.clientX > rect.right ||
+			event.clientY < rect.top ||
+			event.clientY > rect.bottom;
+		if (isOutsideSheet) {
 			onClose();
 		}
 	}
```
[the remark this answers is not in the document]
Comment (Omochice) [reply packages/site/src/lib/CardDetail.svelte:58]: fixed in ae16e6a

File: packages/site/src/lib/CardDetail.svelte [hunk 10]
```diff
@@ -24,142 +25,173 @@
...
+	/*
+	 * Visual dim only. The dialog's own ::backdrop is kept transparent and used
+	 * solely for modal hit-testing, so this element can fade in and out with the
+	 * sheet via a Svelte transition, which ::backdrop cannot do on exit.
+	 */
 	.detail-backdrop {
 		position: fixed;
 		inset: 0;
 		background: rgba(0, 0, 0, 0.4);
 		z-index: 50;
```
[the remark this answers is not in the document]
Comment (Omochice) [reply packages/site/src/lib/CardDetail.svelte]: fixed in 723927b
